lib/ansible/plugins/connection/network_cli.py

```diff
@@ -299,7 +300,7 @@ def update_play_context(self, pc_data):
         play_context.deserialize(pc_data)
 
         self.queue_message('vvvv', 'updating play_context for connection')
-        if self._play_context.become ^ play_context.become:
+        if boolean(self._play_context.become) ^ boolean(play_context.become):
```

Comment:
so this should not be needed as `become` is defined as a boolean attribute, it is an indication that 'validation' is not happening on the play_context objects, which can lead to many of the same issues for other fields.

The proper fix would be to ensure both objects trigger full validation of their attributes.

Reply:
Ok, can you give me a hint where this validation should be done?